src/libraries/System.IO.Hashing/src/System/IO/Hashing/XxHash3.cs

```diff
@@ -0,0 +1,884 @@
+        private static ReadOnlySpan<byte> DefaultSecret => new byte[]
+        {
+            0xb8, 0xfe, 0x6c, 0x39, 0x23, 0xa4, 0x4b, 0xbe, 0x7c, 0x01, 0x81, 0x2c, 0xf7, 0x21, 0xad, 0x1c,
+            0xde, 0xd4, 0x6d, 0xe9, 0x83, 0x90, 0x97, 0xdb, 0x72, 0x40, 0xa4, 0xa4, 0xb7, 0xb3, 0x67, 0x1f,
+            0xcb, 0x79, 0xe6, 0x4e, 0xcc, 0xc0, 0xe5, 0x78, 0x82, 0x5a, 0xd0, 0x7d, 0xcc, 0xff, 0x72, 0x21,
+            0xb8, 0x08, 0x46, 0x74, 0xf7, 0x43, 0x24, 0x8e, 0xe0, 0x35, 0x90, 0xe6, 0x81, 0x3a, 0x26, 0x4c,
+            0x3c, 0x28, 0x52, 0xbb, 0x91, 0xc3, 0x00, 0xcb, 0x88, 0xd0, 0x65, 0x8b, 0x1b, 0x53, 0x2e, 0xa3,
+            0x71, 0x64, 0x48, 0x97, 0xa2, 0x0d, 0xf9, 0x4e, 0x38, 0x19, 0xef, 0x46, 0xa9, 0xde, 0xac, 0xd8,
+            0xa8, 0xfa, 0x76, 0x3f, 0xe3, 0x9c, 0x34, 0x3f, 0xf9, 0xdc, 0xbb, 0xc7, 0xc7, 0x0b, 0x4f, 0x1d,
+            0x8a, 0x51, 0xe0, 0x4b, 0xcd, 0xb4, 0x59, 0x31, 0xc8, 0x9f, 0x7e, 0xc9, 0xd9, 0x78, 0x73, 0x64,
+            0xea, 0xc5, 0xac, 0x83, 0x34, 0xd3, 0xeb, 0xc3, 0xc5, 0x81, 0xa0, 0xff, 0xfa, 0x13, 0x63, 0xeb,
+            0x17, 0x0d, 0xdd, 0x51, 0xb7, 0xf0, 0xda, 0x49, 0xd3, 0x16, 0x55, 0x26, 0x29, 0xd4, 0x68, 0x9e,
+            0x2b, 0x16, 0xbe, 0x58, 0x7d, 0x47, 0xa1, 0xfc, 0x8f, 0xf8, 0xb8, 0xd1, 0x7a, 0xd0, 0x31, 0xce,
+            0x45, 0xcb, 0x3a, 0x8f, 0x95, 0x16, 0x04, 0x28, 0xaf, 0xd7, 0xfb, 0xca, 0xbb, 0x4b, 0x40, 0x7e,
+        };
+
+#if DEBUG
+        static XxHash3()
+        {
+            // Make sure DefaultSecret is the custom secret derived from a seed of 0.
+            byte* secret = stackalloc byte[SecretLengthBytes];
+            DeriveSecretFromSeed(secret, 0);
+            Debug.Assert(new Span<byte>(secret, SecretLengthBytes).SequenceEqual(DefaultSecret));
+
+            // Validate some relationships.
+            Debug.Assert(InternalBufferLengthBytes % StripeLengthBytes == 0);
+        }
+#endif
+
+        private State _state;
+
+        /// <summary>Initializes a new instance of the XXH3 class using the default seed value 0.</summary>
+        public XxHash3() : this(0)
+        {
+        }
+
+        /// <summary>Initializes a new instance of the XXH3 class using the specified seed.</summary>
+        public XxHash3(long seed) : base(HashLengthInBytes)
+        {
+            _state.Seed = (ulong)seed;
+
+            fixed (byte* secret = _state.Secret)
+            {
+                if (seed == 0)
+                {
+                    DefaultSecret.CopyTo(new Span<byte>(secret, SecretLengthBytes));
+                }
+                else
+                {
+                    DeriveSecretFromSeed(secret, (ulong)seed);
+                }
+            }
+
+            Reset();
+        }
+
+        /// <summary>Computes the XXH3 hash of the provided data.</summary>
+        /// <param name="source">The data to hash.</param>
+        /// <returns>The XXH3 64-bit hash code of the provided data, written in little-endian order.</returns>
```

Comment:
Endianness should only be described in the class remarks, if anywhere.  Non-Cryptographic Hash Algorithms produce bytes, not numbers.

```suggestion
        /// <returns>The XXH3 64-bit hash code of the provided data.</returns>
```

You could also leave out 64-bit, if you want, since XXH3 is 64-bit, and XXH128 is the 128-bit variant.

Reply:
Someone needs to know how to turn the bytes back into the same number they would have gotten from the algorithm.  Until we have a public method that returns a number, the only way to do that is by knowing whether you need to use little- or big- endian to interpret the data.  These algorithms hash to numbers and then the shape of our types requires those to be written as bytes.

Reply:
Sure, but for consistency with the other NCHashAlgorithm types, the endianness comment belongs not at the method level, but in the class remarks.

Since I expect the "AsInt64" methods to appear this release it won't really apply to anyone using XxHash3... if they want the number they'll ask for the number, if they want the bytes (e.g. for a network/file member) they'll just use the bytes opaquely.

Reply:
> Sure, but for consistency with the other NCHashAlgorithm types, the endianness comment belongs not at the method level, but in the class remarks.

I don't mind also adding it to the class remarks as well, but I don't see why it shouldn't be on the method.  The method is what's writing out the bytes, not the class.

Either the endianness doesn't matter when writing out the bytes, in which case I don't understand your previous objection to using little-endian, or the endianness does matter, in which case we should be clear when writing out the bytes what endianness is being used.  And if someone is using the derived type in a strongly-typed manner, they know the algorithm being used, and as you pointed out, the specification is very clear that a number is being produced and then translated into big-endian-ordered bytes, so why shouldn't the method doing that exact work be clear as to the work it's doing?

If the other derived types don't document this on the hash methods, seems like they should.

Reply:
This comes back to my notes on the AsInt64 proposal:  hash algortihms, cryptographic or otherwise, produce bytes (not shorts/ints/longs/int128s... bytes).  The internal workings of this algorithm (and, indeed, most hash algorithms) use numbers, but in the end, it's bytes.

A consumer of an NCHashAlgorithm gets bytes.  The bytes they get are the bytes we give them.  If there's reasonable room for debate (e.g. the algorithm it's based is described as producing a number, but we've picked an endianness) then that's a class remark, but the hash methods always produce bytes, opaquely.  For our XxHash32-outputting-as-binary, the spec chose big endian as the answer, so XxHash32 doesn't need to defend its position, so it doesn't have a class remark on the endianness... it's just what the algorithm does.

The AsInt64/etc methods can then describe their relationship to the bytes, but the bytes are king.  Even if the algorithm implementation disagrees (e.g. the pending-public here being the  real worker, and the Hash methods exporting that data).

Reply:
It's not just the algorithm implementation disagreeing... it's the algorithm disagreeing.  The algorithm produces a number.  Anyone familiar with the algorithm knows it produces a number.  Practically every other implementation of the algorithm I can find produces a number.  The description of the algorithm talks about it producing a number, to the point where the spec has the call out you cited that states if you must get bytes here's how to get them from the number.  So if someone has an XxHash3 class and then call a method that gives them bytes, that method is by the definition of this being the XXH3 algorithm getting a number and converting it to bytes; the method description should state how that conversion is being done.  I do not understand the aversion to including that in the method description.